### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by Guy-Gustave**

### CODE REVIEW

**Review by elmejdki — CHANGES_REQUESTED** [review]
# Great job @Guy-Gustave so far, I will not be able to review your code untill you make your repo looks professional.
### status: require changes :recycle: 

Friendly reminder: You should make sure that your repo looks professional before review.
Please take a step back and read a lesson here https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/.
Then apply 3 simple rules from this lesson (it will be easy as you can use a template provided there).
This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!

### your pull request is messing a meaningful title and short summary, please add them and request another code review.
## Happy Coding :tophat:

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello @Guy-Gustave** 
Good job with your new submission for this project :muscle:  
Your page passes most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:
- [ ] Check HTML markup with this validator https://validator.w3.org/#validate_by_upload By now the validator lint some errors in your markup.
- [ ] the search in your navbar should be an input with a search icon.
- [ ] your page has to change with the same RESPONSIVE WEN DESIGNS breakpoints of the original page.
- [ ] for the small screen size the original page shows a user icon in the header, check out:



Your page doesn't have this icon:



Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by Guy-Gustave** [comment]
thanks for your suggestions. now I made all
Screenshot from 2020-06-25 14-44-56

File: README.md
Comment (elmejdki) [inline README.md]: Remove this line if you don't have a reason to keep it

File: README.md
Comment (elmejdki) [inline README.md]: Remove this line if you don't have a reason to keep it

File: index.html
Comment (elmejdki) [inline index.html]: remove the flex column it's ruining the top header.

File: index.html
Comment (elmejdki) [inline index.html]: why are you using col????, please remove it it's unnecessary.

File: index.html
Comment (elmejdki) [inline index.html]: center the text inside the li's, cause they are not centered, you can use text-center:
image

**Review by elmejdki — CHANGES_REQUESTED** [review]
# Good job so far @Guy-Gustave :clap:, but in order to make your project approved you need to make some changes. :hammer: :hammer_and_wrench: :toolbox: 
## status: require changes. :recycle: 

- [ ] fix the header please it not like the original page, they should be exactly the same:
### clone:
image

### original:
image

- [ ] Remove the unnecessary padding from the right and left of the top header on all screen sizes:
### desktop:
image

### tablet:
image

- [ ] you need to hide the navbar on mobile versions:
image

- [ ] the header should be fixed on the top, there is no header in your page:
image

- [ ] Fix the story section:
### clone:
image
### original:
image

- [ ] Fix in the magazine section, fix the title and the articles one by one:
### clone:
image

### original:
image

- [ ] Fix the Editor Pick's section, fix the title and the articles:
### clone:
image

### original:
image

 
- [ ] Fix the feature slideshow:
### clone:
image

### original:
image


- [ ] The subscribe section looks terrible please fix it:
### clone:
image
### original:
image


- [ ] Fix the footer please, it not like the original at all:
### clone: 
image

### original:
image

**Comment by Guy-Gustave** [comment]
i removed the second navbar and all title also the section before the footer. thankx i hope now all is okay

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
****
### Hi @Guy-Gustave ! 👋 
You have done a great job implementing the changes requested by the previous TSE! 👍 You have also done a wonderful job with your project so far! It looks really good! 👏 👏 👏  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- **General**

- [ ] A horizontal scroll appears on your project in both the large and small screen sizes. Please fix this. 
Small screen 👇 
Screen Shot 2020-06-25 at 7 03 24 PM
Large Screen 👇 
Screen Shot 2020-06-25 at 7 01 52 PM


- **Navigation Bar**

- [ ] There is still no input with a search icon. This was requested by a previous TSE. Please fix this.
Your clone 👇 
Screen Shot 2020-06-25 at 7 05 40 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 06 09 PM

- [ ] As was requested by the previous TSE, your navigation bar should look more similar to the original page. The background of the login should not be white. Please fix this. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 09 23 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 09 39 PM

- [ ] The links just below the read header should be centered. Yours are aligned to the right. Please fix this.
Your clone 👇 
Screen Shot 2020-06-25 at 7 11 30 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 11 46 PM

- [ ] Your navigation bar is not responsive on the smaller screen size. Please compare your clone with the original page in the screenshots below and fix this. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 13 26 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 14 39 PM

- [ ] The previous reviewer requested that your header remain fixed at the top of the page, but you have not yet done this. Please implement this requested feature. 


- **Stories sections**
 

- [ ] The top story is missing the article image tag. Please compare your clone with the original page and fix this. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 18 49 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 19 13 PM

- [ ] The image tags on all of the other images are not properly positioned. Please see a few screenshots of the issue below. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 20 39 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 21 00 PM
Your clone 👇 
Screen Shot 2020-06-25 at 7 21 32 PM

The original page 👇 
Screen Shot 2020-06-25 at 7 21 49 PM

- [ ] These elements should take up more of the width of the page. Please fix this. 
Screen Shot 2020-06-25 at 7 25 36 PM


- **In the magazine section**

- [ ] The previous reviewer requested that you improve the titles and articles. Please compare your clone to the original and fix this.
The original page 👇 
Screen Shot 2020-06-25 at 7 27 57 PM
Your clone 👇 
Screen Shot 2020-06-25 at 7 28 15 PM

- [ ] On the medium screen size, the article tags are overlapping. Please fix this.
Your clone 👇 
Screen Shot 2020-06-25 at 7 29 31 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 29 49 PM

- [ ] The articles should take up more of the width of the page in the small screen size. 
The original page 👇 
Screen Shot 2020-06-25 at 7 30 58 PM
Your clone 👇 
Screen Shot 2020-06-25 at 7 31 21 PM

- **Editor's Picks section**

- [ ] The article tags are not properly aligned. Please fix this. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 32 15 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 32 38 PM

- [ ] On the medium screen and small screen size, the articles should have less padding on either side. 
The original page 👇 
 _(medium)_
Screen Shot 2020-06-25 at 7 33 39 PM
_(small)_
Screen Shot 2020-06-25 at 7 35 10 PM

Your clone 👇 
 _(medium)_
Screen Shot 2020-06-25 at 7 33 55 PM
_(small)_
Screen Shot 2020-06-25 at 7 35 34 PM



- **Featured slideshow section**

- [ ] The article images listed on the side are not big enough. Please fix this as was requested by the previous TSE. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 37 24 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 37 46 PM

- [ ] On the medium screen, the images shrink to a size that is way too small. Please fix this. 
Your clone 👇 
Screen Shot 2020-06-25 at 7 38 37 PM
The original page 👇 
Screen Shot 2020-06-25 at 7 38 58 PM


- **Subscribe & Newsletter section**

- [ ] The previous reviewer requested that you fix this section. You have not appeared to have done this. Please compare your clone with the original and fix this. It does not at all look similar to the original. 
The original page 👇 
_(Large)_
Screen Shot 2020-06-25 at 7 41 25 PM
_(Medium)_
Screen Shot 2020-06-25 at 7 43 17 PM

_(Small)_
Screen Shot 2020-06-25 at 7 43 54 PM
Screen Shot 2020-06-25 at 7 44 23 PM

Your clone 👇
_(Large)_ 
Screen Shot 2020-06-25 at 7 41 47 PM
_(Medium)_
Screen Shot 2020-06-25 at 7 43 02 PM

_(Small)_
Screen Shot 2020-06-25 at 7 43 35 PM

- **Footer**

- [ ] The footer also does not look like the original. This was also pointed out to you by the previous TSE. Please fix this. 
Your clone 👇
_(Large)_ 
Screen Shot 2020-06-25 at 7 45 59 PM

_(Medium)_
Screen Shot 2020-06-25 at 7 46 15 PM

_(Small)_
Screen Shot 2020-06-25 at 7 46 25 PM

The original page 👇
_(Large)_ 
Screen Shot 2020-06-25 at 7 46 37 PM

_(Medium)_
Screen Shot 2020-06-25 at 7 46 49 PM

_(Small)_
Screen Shot 2020-06-25 at 7 46 59 PM








****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

**Review by christianotieno — CHANGES_REQUESTED** [review]
# **CHANGES REQUIRED** :warning:
Hey @Guy-Gustave :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, still do have some few changes to carry out before we proceed (see review comment).
##
Kindly take note of the differences below 👇 and work on the suggested changes.

## Main Section Changes

|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/Guy-Gustave/newsweek/feature-branch/index.html)|
|:---:|:---:|
|source|clone|

- Improve on the styling of the Newsweek logo, the login and the subscribe buttons as well. Make sure the Newsweek logo gets centred in the navbar as well. 
- Remove the black borders on all the section articles on the page. 
- Centre the entire contents of the page just as in the source page.

## In the magazine section
|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/Guy-Gustave/newsweek/feature-branch/index.html)|
|:---:|:---:|
|in-the-magazine|in-the-magazine-clone|

- Let's work on the differences highlighted above in this section. From the fonts to the spacing of both the headers articles and the articles themselves.

## Editors pick
|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/Guy-Gustave/newsweek/feature-branch/index.html)|
|:---:|:---:|
|editors-pick|editors-pick-clone|
- Improve on the styling of the editors pick, kindly.
## US, World, etc section
|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/Guy-Gustave/newsweek/feature-branch/index.html)|
|:---:|:---:|
|us|us-clone|
- Remove the black border on all the articles and make sure the height of the rightmost articles are increased to match the source page. 
## Footer Section 
|[Source Page](https://www.newsweek.com/)|[Clone Page](https://raw.githack.com/Guy-Gustave/newsweek/feature-branch/index.html)|
|:---:|:---:|
|footer|footer-clone|
- Improve on the styling on the footer.
## 

### Kindly work on the responsiveness as well. So far there is som much responsiveness with the page.
##
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
- When you have made the specified changes, please submit another code review. Thank you!
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required  🏗️ 🔨 
****
### Hello @Guy-Gustave   👋 
You are doing a great job with this project! 👍 You are almost done! 💪 Great job applying the feedback you have received from myself and the other TSEs. 🎖️ 🌟 There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- [ ] There are no linter checks available in your repo. Please ensure that stickler is enabled. 

- [ ] Your login button is still not the same as the original. Please note that it should be the same background color as the header with white text. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 17 29 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 17 50 AM

- [ ] Your header needs to be improved on the **small screen**. In your clone, the logo overlaps with the other elements of the navigation. Additionally, there is too much empty space on the right, the header should take the full width of the page. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 19 44 AM
 The original page 👇 
Screen Shot 2020-06-27 at 12 20 40 AM

- [ ] The side articles in the featured slideshow section on the **medium screen** are still way too small. Please fix this. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 25 39 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 26 30 AM

- [ ] Please improve the in the magazine section as was requested by the previous TSE. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 28 09 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 28 37 AM

- Your footer is not at all like the original nor is it responsive. 
**Large screen**

- [ ] The subscribe section should have **two** magazine covers **overlapping**. Additionally, your subscribe button is not formatted correctly and the text should be **left-aligned** not centered. Please fix. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 31 38 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 30 20 AM

- [ ] The image in the newsletter section is covering the text. Additionally, the text should be left-aligned not centered and the image should take the full height of the page. 
Your clone 👇 
Screen Shot 2020-06-27 at 12 33 56 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 34 16 AM
**Medium Screen**

- [ ] The newsletter and subscribe sections should both align to **one** element per row.
Your clone 👇 
Screen Shot 2020-06-27 at 12 36 20 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 36 53 AM

- [ ] In addition to the fact that the newsletter section should have **two overlapping magazine covers**, the text **should not be centered** and the image **should not shrink to a smaller size.**
The original page 👇 
Screen Shot 2020-06-27 at 12 38 06 AM
Your clone 👇 
Screen Shot 2020-06-27 at 12 38 29 AM

- [ ] In the subscribe section, the image in your clone completely covers the text. This should not happen. 
The original page 👇 
Screen Shot 2020-06-27 at 12 39 52 AM
Your clone 👇 
Screen Shot 2020-06-27 at 12 40 14 AM

- [ ] Both the newsletter and the subscribe sections need to be improved on the small screen. Please compare your clone with the original and fix this.
Your clone 👇 
Screen Shot 2020-06-27 at 12 41 29 AM
The original page 👇 
Screen Shot 2020-06-27 at 12 42 05 AM
Screen Shot 2020-06-27 at 12 42 14 AM








****
### Kindly make the required changes and resubmit for a new code review once you are ready 🙏


Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

**Comment by Guy-Gustave** [comment]
after correction of the navbar, articles of the right sidebar and the section before the footer

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Status: Changes required 🛑

## Hello @Guy-Gustave 👋, you have done an incredible job making the changes. 

However, we still need to work on some more issues before we can merge.

### Remember to make *ALL* the changes requested by previous TSEs *BEFORE* requesting for another code review. In addition to that,

- There is a horizontal scrollbar in your clone. Find out what's causing it and fix that.
Annotation 2020-06-27 153729

- The clone must have some space on either side like the original.

#### Clone
image

#### Original
image

- Your clone is breaking in the mobile view.

image

### Please make sure that you have applied the changes suggested by previous TSEs. 

Happy coding 👨‍💻

**Comment by Guy-Gustave** [comment]
I finish making the correction you told me again. i hope now all required i made thank you

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Hello @Guy-Gustave,** I appreciate your work and effort. It looks nice but we need to fix some major issues before moving ahead. As you know this is a clone project and we have to make it very similar to the original site in terms of design layout as well as responsiveness. Therefore, let's put a little bit more effort into it and make it a perfect clone of the original Newsweek's site. We have to focus on alignments of elements, margin, and padding. Currently, It is also not as responsive as we want. Please compare your layout with the original one and make necessary changes especially on the subscribe and newsletter section. Furthermore, It seems like you forgot to enable the stickler. Please enable it and remove errors if any.

**Kindly make the required changes suggested by the TSEs and request for another review.
If you need help, message me on slack Suman Shrestha
happy weekends and coding :slightly_smiling_face:**

**Comment by Guy-Gustave** [comment]
I need another review I hope now all required done, i activated linters again and other suggestions

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Status: Changes required 🛑

## Hello @Guy-Gustave 👋, it's me again. I see you are making an effort and I really appreciate it 👏👏. Just a very few changes needed.

- Your README files contain lines from previous merge conflicts. Please remove them. It looks unprofessional. The previous TSE has also pointed this out.

image

image

- I see that you have made an effort to remove the horizontal scrollbar. But your solution has introduced another bug which is present all over the clone.

Here I have pointed a few.
Annotation 2020-06-27 221420

You have applied `overflow-x: hidden;` to the global selector, which is not a solution, but a band-aid. This is also the reason for random scrollbars appearing on elements. Try to find exactly what element is causing the overflow and then fix that. [Here](https://chrome.google.com/webstore/detail/pesticide-for-chrome/bblbgcheenepgnnajgfpiicnbbdmmooh) is a great tool for debugging these kinds of issues.

- Good thinking on using Bootstrap cards for article containers 👌. But cards come with borders on all sides. The original does not have borders on all sides. They have borders on the bottom (or top, depending on your preference). Remember to remove excess borders to match the original.

#### Clone
Annotation 2020-06-27 222201

#### Original
image

- In `newsletter` section, the heading must be on the left side. Right now it's on the right side.

#### Clone
Annotation 2020-06-27 223023

#### Original
Annotation 2020-06-27 223111

- Also, the image is missing in desktop mode.

#### Clone
image

#### Original
image

- `subscribe` section does not match the original in tablet mode,

#### Clone
image

#### Original
image

- In the mobile view, your `subscribe/newsletter` section is not responsive.

#### Clone
image

#### Original
image

Everything is coming together now. Just a few more to go.

Happy coding 👨‍💻

**Comment by Guy-Gustave** [comment]
after removing all border cards and changer the over-flow by fixing the elements that were flowing, also readme. by many different corrections, I still have hope but I hope now okay. thank you

**Review by HADDADSOHAIB — CHANGES_REQUESTED** [review]
### Status: Change Required :recycle: 

### Hi everyone   :hand: 

### Great project, godd work, you implemented a lot of required changes, but there is some changes to be made:

- These navigation elements should be **links**, which are not in your clone
Screenshot from 2020-06-27 22-18-31
- Please decrease the size of the logo in medium and small width.
clone:
Screenshot from 2020-06-27 22-20-54
original:
Screenshot from 2020-06-27 22-21-43
- Please remove the horizontal scroll bar:
Screenshot from 2020-06-27 22-24-47
- I think this part needs some improvement, the alignment and the size of the image are not ok in the medium widths.
clone:
Screenshot from 2020-06-27 22-28-51
Original:
Screenshot from 2020-06-27 22-29-26
- The newsletter in small widths is broken:
clone:
Screenshot from 2020-06-27 22-33-13
- original:
Screenshot from 2020-06-27 22-33-02
### Once done with the changes, please ask for another code review.

**Comment by Guy-Gustave** [comment]
can you see it again. I hope now okay

**Review by christianotieno — CHANGES_REQUESTED** [review]
### Hello, Gustave. I picked up a couple of things while going through this PR. I noticed how much of a task, the project must have been for you. It must have been overwhelming, but for some reason, you kept on going, and never gave up. Keep up with the spirit. I will, however, share some notes below. Kindly read on them before moving on. 
##
## The page has almost all the sections that are required. Good job with that.
- The styling was a major issue with this project. Most of my colleagues, I included tried raising these issues in this PR. Some of the issues were resolved some not. I would suggest taking some time to go through the changes and figure out how you faired.

## The commit messages 
- Though not clear enough, could always be improved. You ought to have ensured the commits were deceptive.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_commits.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)). Have a look at this [article](https://chris.beams.io/posts/git-commit/) for more about improving on your commit messages.
## Guidelines for sending in a Code Review Request.
- The project, when sent in for a review, was still incomplete. from the dashboard, the rule is to always send in a complete project for review.  That was never the case. let's adhere to this guiding principle next time, please.
## Advice
- Take some time and compare the source page with the clone then work on all the changes step by step. You don't need to rush. After which be sure to validate the markup as well. When you are done and truly believe everything looks strikingly similar to the source page in this assignment then you can send it in for a review. Avoid sending in unfished work kindly. Thank you!

**Review by trekab — CHANGES_REQUESTED** [review]
## Changes Required! :recycle: 

Hi @Guy-Gustave, I have noticed that you haven't implemented all the mandatory feedback from
[this review](https://github.com/Guy-Gustave/newsweek/pull/1#pullrequestreview-438726097). It will be a lot easier and faster if you worked on all the changes before you requested for another code review.

__P.S__: If you feel that you need extra assistance(1 on 1), reach out to a full-time T.S.E. Check out [this article](https://microverse.zendesk.com/hc/en-us/articles/360026625973-How-can-I-get-help-from-TSEs-) about how you can schedule time with a full-time T.S.E

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

**Comment by Guy-Gustave** [comment]
i have tried to make all required and suggested. thank you.html validator, linters, add search icon in navbar.

**Comment by addod19** [comment]
## Pull Request under review by Daniel Larbi Addo (TSE)

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @Guy-Gustave, great job with this project :+1: I appreciate your determination to learn and improve :smile: However, most of the comments by the previous TSE's were not addressed. I would highlight some of them you can get them addressed. Kindly be patient and ensure the comments are addressed before you send another code review :100: Below are the comments
- Our page is not responsive on medium devices, let us address it by comparison. We could compare our project with the original project, [newsweek](https://www.newsweek.com/) so we can fix the responsiveness
- It was mentioned earlier that the navigation links should be clickable links, it has not been addressed, kindly fix it
- Kindly compare the screenshot below and fix the size of the images
``` clone ```
im
``` source ```
im1

Kindly address these comments and submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by Guy-Gustave** [comment]
hello addo, thank you for your suggestion, i cannot give up. i tried to change the navigation links and the size image. and correct some responsivness

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi @Guy-Gustave, congratulations :star: You hard work and dedications has really paid off :100: 

Go ahead and merge to master, do not forget to submit the project completion form :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**
